SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -4037,6 +4094,444 @@ void CEulerSolver::ReadActDisk_InputFile(CGeometry *geometry, CSolver **solver_c
+    if (ActDiskBem_file.fail()) SU2_MPI::Error("Unable to open Actuator Disk BEM Input File", CURRENT_FUNCTION);
+
+    string text_line_appo;
+
+    getline(ActDiskBem_file, text_line_appo);
+    /*--- Read and assign the value of the number of propeller blades. ---*/
+    getline(ActDiskBem_file, text_line_appo);
+    istringstream NBlade_value(text_line_appo);
+    NBlade_value >> ADBem_NBlade;
+
+    /*--- Read and assign the value of the propeller diameter. ---*/
+    getline(ActDiskBem_file, text_line_appo);
+    istringstream Diameter_value(text_line_appo);
+    Diameter_value >> ADBem_Diameter;
+
+    /*--- Read and assign the value of the propeller hub radius. ---*/
+    getline(ActDiskBem_file, text_line_appo);
+    istringstream HubR_value(text_line_appo);
+    HubR_value >> ADBem_HubRadius;
+
+    /*--- Read and assign the value of the blade angle at 75% of R. ---*/
+    getline(ActDiskBem_file, text_line_appo);
+    istringstream Angle75R_value(text_line_appo);
+    Angle75R_value >> ADBem_Angle75R;
+
+    getline(ActDiskBem_file, text_line_appo);
+    /*--- Read and assign the value of the number of radial propeller blade sections and alphas for each. ---*/
+    getline(ActDiskBem_file, text_line_appo);
+    istringstream NSection_NAlpha_value(text_line_appo);
+    NSection_NAlpha_value >> ADBem_NSection >> ADBem_NAlpha;
+
+    /*--- Assign the vector dimensions. ---*/
+    i_v.resize(ADBem_NSection, 0.0);
+    radius_v.resize(ADBem_NSection, 0.0);
+    chord_v.resize(ADBem_NSection, 0.0);
+    angle75r_v.resize(ADBem_NSection, 0.0);
+
+    getline(ActDiskBem_file, text_line_appo);
+    /*--- Read and assign the values of the propeller blade section's id, radius, chord, angle75R. ---*/
+    for (iSection = 0; iSection < ADBem_NSection; iSection++) {
+      getline(ActDiskBem_file, text_line_appo);
+      istringstream sectionStream(text_line_appo);
+      sectionStream >> i_v[iSection] >> radius_v[iSection] >> chord_v[iSection] >> angle75r_v[iSection];
+    }
+
+    /*--- Assign the matrix dimensions. ---*/
+    alpha_m.resize(ADBem_NAlpha, std::vector<su2double>(ADBem_NSection, 0.0));
+    cl_m.resize(ADBem_NAlpha, std::vector<su2double>(ADBem_NSection, 0.0));
+    cd_m.resize(ADBem_NAlpha, std::vector<su2double>(ADBem_NSection, 0.0));
+
+    /*--- Read and assign the values for each of the propeller blade section's alpha, cl, cd. ---*/
+    for (iSection = 0; iSection < ADBem_NSection; iSection++) {
+      getline(ActDiskBem_file, text_line_appo);
+      for (iAlpha = 0; iAlpha < ADBem_NAlpha; iAlpha++) {
+        getline(ActDiskBem_file, text_line_appo);
+        istringstream alphaStream(text_line_appo);
+        alphaStream >> alpha_m[iAlpha][iSection] >> cl_m[iAlpha][iSection] >> cd_m[iAlpha][iSection];
+      }
+    }
+  }
+
+  /*--- Update the propeller load according to the modified flow field after every 40 inner iterations. ---*/
+  if (InnerIter % 40 == 0) {
```

Comment:
Seems like you can simply return early.
```suggestion
  if (InnerIter % 40 != 0) return;
```